File: AmIFunny_Mobile/androidApp/src/main/java/com/teegarcs/amifunny/android/saved/SampleNativeAbstractionImpl.kt
package com.teegarcs.amifunny.android.saved

import com.teegarcs.amifunny.features.save_joke.SampleNativeAbstraction

/**
 * Sample of how to use the abstraction of some native framework without using expects/actuals
 */
class SampleNativeAbstractionImpl : SampleNativeAbstraction {
    override fun loadSomethingNative(completion: (String) -> Unit) {
        //DO SOMETHING NATIVE!
        completion("result")
    }
}

File: AmIFunny_Mobile/iosApp/iosApp/saved/SampleNativeAbstractionImpl.swift
//
//  SampleNativeAbstractionImpl.swift
//  iosApp
//
//  Created by Clinton Teegarden on 2/20/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import shared

/**
 * Sample of how to use the abstraction of some native framework without using expects/actuals
 */
class SampleNativeAbstractionImpl: SampleNativeAbstraction{
    
    func loadSomethingNative(completion: @escaping (String) -> Void) {
        //DO SOMETHING NATIVE!
        completion("Result")
    }
    
}
